apps/examples/secure_storage/secure_storage.c

```diff
@@ -0,0 +1,187 @@
+ * LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ * POSSIBILITY OF SUCH DAMAGE.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <security/security_common.h>
+#include <security/security_api.h>
+#include <tinyara/seclink.h>
+#include <tinyara/security_hal.h>
+#include <security/security_ss.h>
+#define SE_IOTIVITY_SVR_DB 0
+extern int rtl_ss_flash_read(uint32_t address, uint32_t len, uint8_t *data, int en_display);
+static char a[8092];
+static char b[8092];
+static int DATA_LENGTH = 128;
+
+#ifdef CONFIG_AMEBAD_TRUSTZONE
+static uint32_t SLOT0_START_ADDR = 0x1a000;
+static uint32_t SLOT1_START_ADDR = 0x1d000;
+#endif
+#ifdef CONFIG_AMEBALITE_TRUSTZONE
+static uint32_t SLOT0_START_ADDR = 0x24000;
+static uint32_t SLOT1_START_ADDR = 0x27000;
+#endif
+static uint32_t SLOT_SIZE = 0x1000;
+static uint32_t SLOT_AREA_OFFEST = 0x23000;
+
+/****************************************************************************
+ * secure_storage_main
+ ****************************************************************************/
+
+#ifdef CONFIG_BUILD_KERNEL
+int main(int argc, FAR char *argv[])
+#else
+int sstorage_main(int argc, char *argv[])
+#endif
+{
+	printf("Start testing SE Secure Storage!!\n");
+	int length = DATA_LENGTH;
+	memset(a,'A', length);
+	a[length] = '\0';
+
+	printf("Start Security init!!\n");
+	sleep(1);
+	security_handle hnd;
+	security_error res = security_init(&hnd);
+	if (res != SECURITY_OK) {
+		dbg("security_init() Fail\n",__LINE__);
+		sleep(1);
+	}
+	printf("=====================%d==================================================\n",__LINE__);
+
+	sleep(1);
+	security_data data = { (void *)a, length};
+	char ss_path[7];
+
+	printf("Start Secure Storage write!!\n");
+	int slot;
+	for (slot = 0; slot < 4; slot++) {
+		snprintf(ss_path, 7, "ss/%d", slot);
+		memset(a,(0xA0 + slot), length);
+		a[length] = '\0';
+		printf("ss_write_secure_storage() Slot: %d\n",slot);
+		sleep(1);
+		res = ss_write_secure_storage(hnd, ss_path, 0, &data);
+		if (res != SECURITY_OK) {
+			dbg("ss_write_secure_storage() Fail\n",__LINE__);
+			sleep(1);
+		}
+	}
+	printf("=====================%d==================================================\n",__LINE__);
+
+	printf("Start Secure Storage read!!\n");
+	sleep(1);
+
+	for (slot = 0; slot < 4; slot++) {
+		snprintf(ss_path, 7, "ss/%d", slot);
+		printf("ss_read_secure_storage() Slot: %d\n",slot);
+		sleep(1);
+		res = ss_read_secure_storage(hnd, ss_path, 0, &data);
+		if (res != SECURITY_OK) {
+			dbg("ss_read_secure_storage() Fail\n",__LINE__);
+			sleep(1);
+		}
+
+		printf("data.length: %d\n",data.length);
```

Comment:
If failed to read the storage, it just waits 1 seconds, but try to print the contents from `data`.
In this case, invalid data can be printed.
I think we can use `continue` for failure case. (But we should consider, in this case, not printed the data for failure case.)

Reply:
Modified